src/main/java/org/isf/stat/manager/JasperReportsManager.java

```diff
@@ -959,10 +959,9 @@ private void addSubReportsBundleParameters(String jasperFileFolder, String jaspe
 		}
 	}
 
-	private void addReportBundleParameter(String jasperParameter, String jasperFileFolder, String jasperFileName, Map<String, Object> parameters) {
+	private void addReportBundleParameter(String jasperParameter, String jasperFileName, Map<String, Object> parameters) {
 		try {
-			ResourceBundle resourceBundle = ResourceBundle.getBundle(jasperFileFolder + File.separator + jasperFileName, new Locale(GeneralData.LANGUAGE),
-							new UTF8Control());
+			ResourceBundle resourceBundle = ResourceBundle.getBundle(jasperFileName, new Locale(GeneralData.LANGUAGE), new UTF8Control());
```

Comment:
Can't `new Locale(GeneralData.LANGUAGE)` be a static class variable as you can't change the language without restarting OH.   That way it isn't reallocated for every report.   Just a thought ....

Reply:
Hi David, I restored that function to how it was before my last change, it was already like that. I'll move the creation to the beginning then as you suggest. Thank you very much

Reply:
The mainMenu set Locale.setDefault(new Locale(GeneralData.LANGUAGE)), so we can use the default locale

Reply:
is it ok now @dbmalkovsky? can we resolve the conversation?

Reply:
yes